evalml/tests/data_checks_tests/test_invalid_targets_data_check.py

```diff
@@ -314,3 +320,67 @@ def test_invalid_target_data_check_initialize_with_none_objective():
     with pytest.raises(DataCheckInitError, match="Encountered the following error"):
         DataChecks([InvalidTargetDataCheck], {"InvalidTargetDataCheck": {"problem_type": "multiclass",
                                                                          "objective": None}})
+
+
+def test_invalid_target_data_check_regression_problem_nonnumeric_data():
+    X = pd.DataFrame()
+    y_categorical = pd.Series(["Peace", "Is", "A", "Lie"])
+    y_mixed_cat_numeric = pd.Series(["Peace", 2, "A", 4])
+    y_integer = pd.Series([1, 2, 3, 4])
+    y_float = pd.Series([1.1, 2.2, 3.3, 4.4])
+    y_numeric = pd.Series([1, 2.2, 3, 4.4])
+
+    data_check_error = DataCheckError(
+        message=f"Target data type should be floating point for regression type problems.",
+        data_check_name=invalid_targets_data_check_name,
+        message_code=DataCheckMessageCode.TARGET_UNSUPPORTED_TYPE,
+        details={}).to_dict()
+
+    invalid_targets_check = InvalidTargetDataCheck("regression", get_default_primary_search_objective("regression"))
+    assert invalid_targets_check.validate(X, y=y_categorical) == {"warnings": [], "errors": [data_check_error]}
+    assert invalid_targets_check.validate(X, y=y_mixed_cat_numeric) == {"warnings": [], "errors": [data_check_error]}
+    assert invalid_targets_check.validate(X, y=y_integer) == {"warnings": [], "errors": []}
+    assert invalid_targets_check.validate(X, y=y_float) == {"warnings": [], "errors": []}
+    assert invalid_targets_check.validate(X, y=y_numeric) == {"warnings": [], "errors": []}
+
+
+def test_invalid_target_data_check_multiclass_problem_binary_data():
+    X = pd.DataFrame()
+    y_multiclass = pd.Series([1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3] * 25)
+    y_binary = pd.Series([0, 1, 1, 1, 0, 0] * 25)
+
+    data_check_error = DataCheckError(
+        message=f"Target does not have more than two classes, which is required for multiclass classification.",
+        data_check_name=invalid_targets_data_check_name,
+        message_code=DataCheckMessageCode.TARGET_MULTICLASS_NOT_ENOUGH_CLASSES,
+        details={"num_classes": len(set(y_binary))}).to_dict()
+
+    invalid_targets_check = InvalidTargetDataCheck("multiclass", get_default_primary_search_objective("regression"))
```

Comment:
Any reason for `get_default_primary_search_objective("regression")` here?

Reply:
pasta, thanks